File: day1/Python/part1.py
import re
with open("input.txt", "r") as file:
    contents = file.read()

arr = re.split(r'\r?\n',contents)
counter = 0
for i,val in enumerate(arr):
    if i == 0 or val > arr[i-1]:
        counter+=1
print(counter)

File: day1/JS/part1.js
const fs = require('fs');
(function countDepthIncrease() {
    let input = fs.readFileSync('input.txt','utf-8').split(/\r?\n/);
    let arr = Array.from(input);

    function compare(element,index,array){
        if (index == 0){
            return true;
        }
        if (element > array[index-1]){
            return true;
        }
    }

    console.log(arr.filter(compare).length);
})();
